apps/zotonic_core/src/support/z_edge_log_server.erl

```diff
@@ -224,11 +219,24 @@ do_edge_notify(<<"INSERT">>, SubjectId, PredName, ObjectId, EdgeId, Acc, Context
         T2 => [ Edge | maps:get(T2, Acc, []) ]
     }.
 
+%% @doc After edge deletion, check if a dependent resource is connected. If not then the resource
+%% can be deleted. There is a waiting period of an hour so that a resource can be disconnected first
+%% before being connected to another resource.
 maybe_delete_dependent(Id, Context) ->
-    case m_rsc:p_no_acl(Id, is_dependent, Context) of
+    maybe_schedule_dependent_check(Id, 3600, Context).
+
+maybe_schedule_dependent_check(Id, Delay, Context) when is_integer(Id) ->
+    case m_rsc:p_no_acl(Id, <<"is_dependent">>, Context) of
         true ->
             Key = z_convert:to_binary(Id),
-            z_pivot_rsc:insert_task_after(3600, ?MODULE, delete_if_unconnected, Key, [Id], Context);
+            case m_edge:has_subjects(Id, Context) of
+                true -> ok;
+                false ->
+                    case z_pivot_rsc:insert_task_after(Delay, ?MODULE, delete_if_unconnected, Key, [Id], Context) of
+                        {ok, _} -> ok;
+                        {error, _} = Error -> Error
+                    end
+            end;
```

Comment:
Why the additional check on `m_edge:has_subjects` now?

AFAICT when the `delete_if_unconnected` happens this is already accounted for.

Reply:
Because the edge log server does not check - and by checking if before the `delete_if_unconnected` we can prevent inserting (and running) it as a task if the resource still has referrers.